### ISSUE TITLE
createType3Hints fails with ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by riezebosch** [1to9only/Sudoku16x16Explainer#1]
https://github.com/1to9only/Sudoku16x16Explainer/blob/8d35a371728a7e76114f4fc12617ea3288f728a9/diuf/sudoku/solver/rules/unique/UniqueLoops.java#L339

Where does this 7 come from? I try to generate new sudokus while limiting the used techniques `EnumSet.range(SolvingTechnique.HiddenSingle, SolvingTechnique.HiddenQuad)` and walk into this exception regularly.
  
![Screenshot 2021-12-07 at 10 22 04](https://user-images.githubusercontent.com/1427334/145001938-1f5b9d08-b114-4d9a-8446-ded7b0e07745.png)

**Comment by 1to9only** [comment]
The Sudoku16Explainer is a modification of SudokuExplainer written for 9x9 sudokus.

The 7 is a remnant of the 9x9 code, I think 7 is simpy 9-2 where degree=2 for hidden/naked pair. So the bug results from an incomplete conversion (by me!) of the code to 16x16. I've now changed the 7 to 14!

If this fixes the problem, please close.

**Comment by 1to9only** [comment]
To recreate the problem (if using an earlier version of Sudoku16Explainer):

- Start the GUI.

- Make sure UR/UL techniques are enabled:
-> Options
-> Solving Techniques...
-> Unique Rectangle / Loop 4.5+ **<- tick this**
-> click OK

- Load this grid:
```
G...B..HJ...L..A
....I.FOPB......
M.P....NLD.OB.I.
K...JL.PA.......
E..GLB.CK..MPA.F
PFO..I..DHL.....
.KIH.EJ.N....L..
.JM....DE.A..BNI
FLCNGPBEH...MIOD
...IO...BFG.A...
...MD.I.O.ELC..B
O.K.AH.JM....G..
IMFO..DGC.......
.CJ.FO.BGMPEI...
.A.K..PLION..J..
....HJ.IF....CMO
```

- Click F6|Get all hints

**Comment by 1to9only** [comment]
The problem is fixed in the latest Sudoku16Explainer (2022.1.12).
